lib/actions/action-create-card.ts

```diff
@@ -91,6 +91,12 @@ const handler: ActionDefinition['handler'] = async (
 			}),
 		);
 	}
+	const properties = request.arguments.properties;
+
+	if (properties?.payload?.file && properties?.data?.payload?.file) {
```

Comment:
@iamsolankiamit Why would there be both a `properties.payload.file` and a `properties.data.payload.file`? I thought it should only be `request.arguments.properties.payload.file`, correct me if I'm wrong. And I think while `request.arguments.properties.payload` can be optional, we should probably make `request.arguments.properties.payload.file` required if `request.arguments.properties.payload` exists.

Reply:
you are right @joshbwlng we don't need to check `properties.data.payload`. I made that change looking at the sdk. About making file as required may work, so the way file payloads work here is we can name the `file` field anything. the  action facade will pick it up and put that name. So i shouldn't even be checking for the file field.

Reply:
@iamsolankiamit Things are looking better. Could you try the beta package from this PR in a new draft PR in the main repo and see what happens when you upload a file through `action-create-card`? We need to check that (a) file uploads work and (b) what the UI looks like for these contracts. My guess is that the UI won't display the uploaded file as nicely as it does for messages/whispers.

Reply:
I guess it won't show any image, since there is a separate method to get the image.

Reply:
I'll report after testing this

Reply:
@joshbwlng I tested it on our flowdock-archive setup. File upload is working, but the UI doesn't show the uploaded file. Right now attachments are only shown in messages, which uses the `AuthenticatedImage` component to fetch the file based on filename and cards id. 

I can PR the flowdock-archive setup along with PRs to support the testing (flowdock-message contract). 

I am looking into the list view lens to see if we can add image/attachment rendering there. It might even be worth to look if we can support attachments out of the box for all the contracts, if the schema matches `data.payload['file_filed_name].filename`